pkg/hintrunner/zero/zerohint_dictionaries.go

```diff
@@ -33,3 +35,56 @@ func newSquashDictInnerAssertLenKeysHint() hinter.Hinter {
 func createSquashDictInnerAssertLenKeysHinter() (hinter.Hinter, error) {
 	return newSquashDictInnerAssertLenKeysHint(), nil
 }
+
+// SquashDictInnerNextKey hint retrieves the next key for processing during
+// dictionary squashing after checking that the array of keys is not empty
+//
+// `newSquashDictInnerNextKeyHint` takes 1 operander as argument
+//   - `next_key` variable will be assigned to the next key in `keys`
+func newSquashDictInnerNextKeyHint(nextKey hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "SquashDictInnerNextKey",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> assert len(keys) > 0, 'No keys left but remaining_accesses > 0.'
+			//> ids.next_key = key = keys.pop()
```

Comment:
I'm wondering if adding the updated value of ```key``` to the scope isn't neccessary, as [this hint](https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/squash_dict.cairo#L206) uses it as key in the access_indices structure. Since ```squash_dict_inner``` is invoked recursively it might be neccessary.

Reply:
of `keys` you mean? yeah its necessary because of the recursive call, which will empty the array one element after the other. If we dont, we'll get an infinite loop

Reply:
oh yeah!!!!! sorry, `key` indeed, is used where you mentioned, i need to add it to the scope! thanks a lot!!!!

Reply:
nice catch!